### DISCUSSION TITLE
Dataset inheritance (based on another dataset)

### DISCUSSION THREAD

**Discussion by GeoffreyOnRails** [apache/superset#25185] [Ideas]
I have a few virtual datasets that are quite complex. To deal with that complexity, I do a few "WITH" SQL clauses, which allows me to have explicit cleaned data.
However, I have some duplication in virtual datasets, as I need the same preprocessing in different charts.
I'd like to have the possibility to base a virtual dataset on another virtual dataset, so that when I change some preprocessing, multiple virtual datasets are affected.

Why can't I create SQL views instead? I could, but the data team has a read-only access to a replica database, so creating views would be less convenient, as there is a review process for everything that goes on a production database. But that's definitely our current workaround for my use-case (just a little less convenient).

**Comment by vergenzt** [comment]
IMO it'd be great to to basically do virtual datasets by reference! I.e. you could make a Jinja macro available to incorporate the _text_ of (the SQL query of) another virtual dataset into the current dataset. Circular dependencies would have to be prevented, but I think it could be a low-effort way to do this. It wouldn't improve performance at all compared to status quo but it would improve maintainability.
